### PR TITLE
Core/Packets: Added CMSG_CLOSE_INTERACTION and CMSG_GOSSIP_SELECT_OPTION

### PR DESCRIPTION
**Pull request by gegge**

### CODE REVIEW

**Comment by gegge** [comment]
ok 1 thing:
How should i remove all these "merge"?
CMSG_GOSSIP_SELECT_OPTION, credit to projectcoreDevs, i only ported the opcode
CMSG_CLOSE_INTERACTION is my first opcode structure with the new format (that's why i looked at projectcore structure's one), and i didn't really know what there is to handle, because it is only a windows that have to close, but now at least there isn't spam in the console..

File: src/server/game/Handlers/NPCHandler.cpp [hunk 1]
```diff
@@ -345,6 +345,18 @@ void WorldSession::HandleGossipHelloOpcode(WorldPackets::NPC::Hello& packet)
...
+void WorldSession::HandleCloseInteractionOpcode(WorldPackets::NPC::CloseInteraction& packet)
+{
+    TC_LOG_DEBUG("network", "WORLD: CMSG_CLOSE_INTERACTION");
+    Creature* unit = GetPlayer()->GetNPCIfCanInteractWith(packet.UnitGuid, UNIT_NPC_FLAG_NONE);
+    if (!unit)
+    {
+        TC_LOG_DEBUG("network", "WORLD: HandleCloseInteractionOpcode - %s not found, nothing to handle.", packet.UnitGuid.ToString().c_str());
+        return;
+    }
+
```
Comment (Intel) [inline src/server/game/Handlers/NPCHandler.cpp]: I don't think we add opcodes to the core that have no handling

File: src/server/game/Server/Protocol/Opcodes.cpp [hunk 2]
```diff
@@ -316,7 +317,7 @@ void OpcodeTable::Initialize()
     DEFINE_OPCODE_HANDLER_OLD(CMSG_GMTICKET_UPDATETEXT,                     STATUS_UNHANDLED, PROCESS_THREADUNSAFE, &WorldSession::HandleGMTicketUpdateOpcode      );
     DEFINE_OPCODE_HANDLER_OLD(CMSG_GM_REPORT_LAG,                           STATUS_UNHANDLED, PROCESS_THREADUNSAFE, &WorldSession::HandleReportLag                 );
     DEFINE_HANDLER(CMSG_GOSSIP_HELLO,                                       STATUS_LOGGEDIN,  PROCESS_THREADUNSAFE, WorldPackets::NPC::Hello, &WorldSession::HandleGossipHelloOpcode);
-    DEFINE_OPCODE_HANDLER_OLD(CMSG_GOSSIP_SELECT_OPTION,                    STATUS_UNHANDLED, PROCESS_THREADUNSAFE, &WorldSession::HandleGossipSelectOptionOpcode  );
+    DEFINE_HANDLER(CMSG_GOSSIP_SELECT_OPTION, STATUS_LOGGEDIN, PROCESS_THREADUNSAFE, WorldPackets::NPC::PlayerCliGossipSelectOption, &WorldSession::HandleGossipSelectOptionOpcode);
```
Comment (Intel) [inline src/server/game/Server/Protocol/Opcodes.cpp]: Align properly

File: src/server/game/Handlers/NPCHandler.cpp (hunk 1 above)
Comment (gegge) [inline src/server/game/Handlers/NPCHandler.cpp]: i think there is nothing to handle there.. i remove that

File: src/server/game/Handlers/MiscHandler.cpp [hunk 3]
```diff
@@ -91,53 +92,46 @@ void WorldSession::HandleRepopRequestOpcode(WorldPacket& recvData)
...
+    if (!_player->PlayerTalkClass->GetGossipMenu().GetItem(packet.GossipID))
     {
-        recvData.rfinish();
         return;
     }
 
-    if (_player->PlayerTalkClass->IsGossipOptionCoded(gossipListId))
-        recvData >> code;
+    if (_player->PlayerTalkClass->IsGossipOptionCoded(packet.GossipID))
+        packet.PromotionCode;
```
Comment (Intel) [inline src/server/game/Handlers/MiscHandler.cpp]: Makes no sense. If it's not needed for parsing then remove this completely

File: src/server/game/Server/Packets/NPCPackets.h [hunk 4]
```diff
@@ -26,6 +26,29 @@ namespace WorldPackets
...
+        public:
+            CloseInteraction(WorldPacket&& packet) : ClientPacket(CMSG_CLOSE_INTERACTION, std::move(packet)) { }
+
+            void Read() override;
+
+            ObjectGuid UnitGuid;
+        
+        };
+
+        class PlayerCliGossipSelectOption final : public ClientPacket
```
Comment (Intel) [inline src/server/game/Server/Packets/NPCPackets.h]: We use JAM names without prefixes. That means the "PlayerCli" part should be removed

File: src/server/game/Handlers/NPCHandler.cpp (hunk 1 above)
Comment (Intel) [inline src/server/game/Handlers/NPCHandler.cpp]: Do not remove yet, it's better to find out what is needed to be handled. Or I believe there is a special opcode table flag that marks opcode as not needed and ignore any errors.

**Comment by gegge** [comment]
Ok for now i corrected everything.

The WoWPacketParser gave me this about the opcode i tried to code:

ClientToServer: CMSG_CLOSE_INTERACTION (0x1B19) Length: 13 ConnIdx: 1 EP:  Time: 12/28/2014 17:41:38.360 Number: 8616
Guid: Full: 0x1C147400200D214000003C00001A407A Creature/0 R1309/S60 Map: 1 Entry: 13445 Low: 1720442

**Comment by Intel** [comment]
I don't think there is an easy way to remove these merge commits. What you have to do is create a new branch based on TrinityCore/6.x and cherry pick the required commits. Or generate a diff and apply on new branch.

**Comment by gegge** [comment]
what is the command to create a branch based on another repo? 
for this branch i did "git checkout -b test"

**Comment by Intel** [comment]
Firstly, you need to set up a tc remote if you don't have it yet (I named it "tc", this name will appear in later commands):
git remote add tc https://github.com/TrinityCore/TrinityCore.git

Fetch all changesets:
git fetch tc

Create branch:
git branch -b tc/6.x

**Comment by gegge** [comment]
i'm going to rewrite this on a clean tc repo, if you want to close you can.. I will update as soon as possible to me

**Pull request closed without merging**